cpp/mcap/include/mcap/writer.inl

```diff
@@ -331,6 +331,7 @@ void McapWriter::open(IWritable& writer, const McapWriterOptions& options) {
       chunkWriter->resetCrc();
     }
   }
+  writer.crcEnabled = compression_ == Compression::None && !options.noCRC;
```

Comment:
CRC is not compatible with compression?

Reply:
Didn't want to both with double-CRC when compression is enabled (compressed chunks already do their own crc). I think we've had some discussions in the past about only computing dataSectionCRC if chunking/compression is not enabled? Or maybe we need to add more writer options? Open to suggestions on what to do here.

Reply:
There's stuff outside of the chunks that needs to be covered by a CRC ie. `MessageIndex`, `Metadata` records, and the fields of the `Chunk` record that aren't the `records` field. I think this should just read:
```
writer.crcEnabled =  !options.noCRC;
```